src/Avalonia.Base/Media/Color.cs

```diff
@@ -382,76 +381,64 @@ private static bool TryParseCssFormat(string s, out Color color)
 
             if (components.Length == 3) // RGB
             {
-                if (InternalTryParseByte(components[0], out byte red) &&
-                    InternalTryParseByte(components[1], out byte green) &&
-                    InternalTryParseByte(components[2], out byte blue))
+                if (InternalTryParseByte(components[0].AsSpan(), out byte red) &&
+                    InternalTryParseByte(components[1].AsSpan(), out byte green) &&
+                    InternalTryParseByte(components[2].AsSpan(), out byte blue))
                 {
                     color = new Color(0xFF, red, green, blue);
                     return true;
                 }
             }
             else if (components.Length == 4) // RGBA
             {
-                if (InternalTryParseByte(components[0], out byte red) &&
-                    InternalTryParseByte(components[1], out byte green) &&
-                    InternalTryParseByte(components[2], out byte blue) &&
-                    InternalTryParseDouble(components[3], out double alpha))
+                if (InternalTryParseByte(components[0].AsSpan(), out byte red) &&
+                    InternalTryParseByte(components[1].AsSpan(), out byte green) &&
+                    InternalTryParseByte(components[2].AsSpan(), out byte blue) &&
+                    InternalTryParseDouble(components[3].AsSpan(), out double alpha))
                 {
                     color = new Color((byte)Math.Round(alpha * 255.0), red, green, blue);
                     return true;
                 }
             }
 
             // Local function to specially parse a byte value with an optional percentage sign
-            bool InternalTryParseByte(string inString, out byte outByte)
+            bool InternalTryParseByte(ReadOnlySpan<char> inString, out byte outByte)
             {
                 // The percent sign, if it exists, must be at the end of the number
-                int percentIndex = inString.IndexOf("%", StringComparison.Ordinal);
+                int percentIndex = inString.IndexOf("%".AsSpan(), StringComparison.Ordinal);
 
                 if (percentIndex >= 0)
                 {
-                    var result = double.TryParse(
-                        inString.Substring(0, percentIndex),
-                        NumberStyles.Number,
-                        CultureInfo.InvariantCulture,
+                    var result = inString.Slice(0, percentIndex).TryParseNumberToDouble(
```

Comment:
I'm not a big fan of the new `TryParseNumberToDouble` type methods here:

 1. It assumes invariant culture and number styles (which I guess that's why the name is so long).
 2. Using the actual double.TryParse() is so much cleaner and more standards-friendly. It's clear culture is being handled correctly.
 3. The new methods don't actually do anything right unless compiled for .net versions that use spans. In .net standard you are actually going string -> span -> string. This is why I really didn't do this before.

Reply:
I would really just add a new version of double.TryParse() that supports span and .net standard 2.0 then use it here. Mentioned: https://github.com/AvaloniaUI/Avalonia/pull/9215/files#r997344774. That makes the code more future-proof, readable and culture-aware IMO.

Reply:
About #3, I would actually say that we don't need to overcomplicate task just for netstandard2.0.
Majority of supported platforms will use .NET 6 or higher. Where netstandard2.0 is kept for small percent of .NET Framework apps build with Avalonia (mostly, as an integration with older framework apps) and some exotic Mono devices.

Reply:
1-2. This is easier to read and should help avoid culture issues now that new parameters are added for number style and culture. Thanks.
3. I don't know the platforms or applications netstandard 2.0 is used for so if you think it's fine I won't disagree.